roles/lib_utils/build/ansible/yedit.py

```diff
@@ -0,0 +1,147 @@
+#pylint: skip-file
+
+def get_curr_value(invalue, val_type):
+    '''return the current value'''
+    if invalue == None:
+        return None
+
+    curr_value = invalue
+    if val_type == 'yaml':
+        curr_value = yaml.load(invalue)
+    elif val_type == 'json':
+        curr_value = json.loads(invalue)
+
+    return curr_value
+
+def parse_value(inc_value, vtype=''):
+    '''determine value type passed'''
+    true_bools = ['y', 'Y', 'yes', 'Yes', 'YES', 'true', 'True', 'TRUE', 'on', 'On', 'ON', ]
+    false_bools = ['n', 'N', 'no', 'No', 'NO', 'false', 'False', 'FALSE', 'off', 'Off', 'OFF']
+
+    # It came in as a string but you didn't specify value_type as string
+    # we will convert to bool if it matches any of the above cases
+    if isinstance(inc_value, str) and 'bool' in vtype:
+        if inc_value not in true_bools and inc_value not in false_bools:
+            raise YeditException('Not a boolean type. str=[%s] vtype=[%s]' % (inc_value, vtype))
+    elif isinstance(inc_value, bool) and 'str' in vtype:
+        inc_value = str(inc_value)
+
+    # If vtype is not str then go ahead and attempt to yaml load it.
+    if isinstance(inc_value, str) and 'str' not in vtype:
+        try:
+            inc_value = yaml.load(inc_value)
+        except Exception as _:
+            raise YeditException('Could not determine type of incoming value. value=[%s] vtype=[%s]' \
+                                 % (type(inc_value), vtype))
+
+    return inc_value
+
+# pylint: disable=too-many-branches
+def main():
```

Comment:
Change to use the unit testing pattern and include the current set of yedit unit tests.